packages/shared/lib/core/profile-manager/api/events/transaction-progress.event.ts

```diff
@@ -0,0 +1,83 @@
+import { selectedAccountId } from '@core/account'
+import { get } from 'svelte/store'
+import { isLedgerProfile } from '@core/profile'
+import { closePopup, openPopup } from '@lib/popup'
+import { ledgerDeviceStatus, ledgerMintNativeTokenConfirmationProps, ledgerSendConfirmationProps } from '@core/ledger'
+
+import { TransactionProgressPayload } from '../types'
+
+export function handleTransactionProgressEvent(accountId: string, payload: TransactionProgressPayload): void {
+    if (get(isLedgerProfile)) {
+        if (get(selectedAccountId) === accountId) {
+            handleTransactionProgressInternal(payload)
+        }
+    } else {
+        console.warn('Transaction progress handler unimplemented: ', payload)
+    }
+}
+
+function handleTransactionProgressInternal(
+    payload: TransactionProgressPayload,
+    isDuringOnboarding: boolean = false
+): void {
+    const sendConfirmationProps = get(ledgerSendConfirmationProps)
+    const mintNativeTokenConfirmationProps = get(ledgerMintNativeTokenConfirmationProps)
+
+    const _openPopup = (type = 'ledgerTransaction') => {
+        const shouldPreventClose = type === 'enableLedgerBlindSigning' || type === 'ledgerTransaction'
+
+        const isBlindSigningTransaction = typeof payload !== 'string' && 'PreparedTransactionEssenceHash' in payload
+        const toAddress = isBlindSigningTransaction
+            ? null
+            : sendConfirmationProps.recipient.type === 'address'
+            ? sendConfirmationProps.recipient.address
+            : sendConfirmationProps.recipient.account.depositAddress
+        const toAmount = isBlindSigningTransaction
+            ? null
+            : `${sendConfirmationProps.amount} ${sendConfirmationProps.unit}`
+        const hash = isBlindSigningTransaction ? payload?.['PreparedTransactionEssenceHash'] : null
```

Comment:
This is an exception to the rule, since I can see in one glance what's happening, but don't have to follow a whole branch of `?` `:`. Maybe a coding convention should be put in place that this is only allowed for one-liners.